RebbleOS/vibrate.c

```diff
@@ -23,18 +23,37 @@
 #include "vibrate.h"
 #include "task.h"
 #include "semphr.h"
+#include <stdbool.h>
 
+#define VIBRATE_QUEUE_MAX_WAIT_TICKS  20    // chosen by fair dice roll
 
-static uint16_t VIBRO_SHORT[] = { 255, 1000 };
-static uint16_t VIBRO_TAP_TAP[] = { 255, 750, 0, 750, 255, 750 };
+
+/*
+ * Initialization of default patterns.
+ * 
+ * TODO Maybe load these dinamically at some point
+ */
```

Comment:
You are completely right, I've changed the buffer type from a `uint16_t` array to an array of structs of type `VibratePatternPair_t` (there's probably a better name for this). 
Each one of these `VibratePatternPair_t` structs contains a duration (in milliseconds) and a frequency at which the motor runs.